t/t3600-rm.sh

```diff
@@ -137,8 +137,8 @@ test_expect_success 'Re-add foo and baz' '
 test_expect_success 'Modify foo -- rm should refuse' '
```

Comment:
Since you heeded @chriscool's advice to focus on only one test script, maybe it would make more sense to use the one-line "t3600: use test_path_is_dir and test_path_is_file"?

Another thing to mention in the commit message is the "why?"... So far, a casual reader will not exactly understand what the benefit might be, and might even disagree that it is an improvement because

1. the new code will be slower (as it adds one level of indirection: a shell function)
2. the new code is more verbose (`test -d` is shorter than `test_path_is_dir`)

Obviously, the active Git developers do agree, though, that it is a good change (otherwise they would not have suggested it as a GSoC microproject), and I think it is because:

1. the new code is a lot more obvious to developers who are not fluent in Unix shell scripting, and
2. the new code is a lot more informative in case of a breakage.